docker-swarm-entrypoint.sh

```diff
@@ -1,38 +1,100 @@
 #!/bin/bash
+function initialize() {
+    echo "Initializing $1 ZooKeeper."
+}
+
 function healthy() {
+    echo -n "Checking health. "
     HEALTHY=$(cat /usr/local/bin/HEALTHY)
 }
 
 function discoverNodes() {
-    echo "Discovering other nodes in cluster..."
-    NODES=$(dig tasks.$SERVICE_NAME +short)
+    echo -n "Discovering nodes in cluster. "
+    NODES=($(dig tasks.$SERVICE_NAME +short))
 }
 
-if [[ -z $SERVICE_NAME ]]
-then
-    echo "Running standalone."
-else
-    echo "Running in Swarm mode for service: $SERVICE_NAME"
+if [[ -n $SERVICE_NAME ]]
+then initialize "replicated"
+else initialize "standalone"
+fi
+
+echo "Container is initializing."
+echo "Setting INITIALIZED=0."
+echo 0 | tee /usr/local/bin/INITIALIZED 1>/dev/null
+
+CONFIG="$ZOO_CONF_DIR/zoo.cfg"
+
+{
+    [[ -z $SERVICE_NAME ]] && echo "clientPort=$ZOO_PORT"
 
+    echo "dataDir=$ZOO_DATA_DIR"
+    echo "dataLogDir=$ZOO_DATA_LOG_DIR"
+
+    echo "tickTime=$ZOO_TICK_TIME"
+    if [[ -n $SERVICE_NAME ]]
+    then
+        echo "initLimit=$ZOO_INIT_LIMIT"
+        echo "syncLimit=$ZOO_SYNC_LIMIT"
+    fi
+
+    echo "maxClientCnxns=$ZOO_MAX_CLIENT_CNXNS"
+    echo "reconfigEnabled=$ZOO_RECONFIG_ENABLED"
+    echo "skipACL=$ZOO_SKIP_ACL"
+} >> "$CONFIG"
+
+healthy
+while [[ $HEALTHY -eq 0 ]]
+do
+    echo "Sleeping for 1s."
+    sleep 1
     healthy
-    while [[ $HEALTHY -eq 0 ]]
-    do
-        sleep 1
-        healthy
-    done
+done
+echo "Healthy."
+
+ZOO_MY_IP=$(hostname -i)
+echo "My IP: $ZOO_MY_IP"
+
+if [[ -f $ZOO_DATA_DIR/myid ]]
+then ZOO_MY_ID=$(cat $ZOO_DATA_DIR/myid)
```

Comment:
@chatsatourian 

having a fixed ID (in case `$ZOO_DATA_DIR/myid` is mounted on the host) won't work cause the Leader is always calculating the IDs of new Followers by their IP in the `zookeeper-cleanup.sh`

Reply:
Fixed.